channeldb/graph.go

```diff
@@ -1798,34 +1798,52 @@ func (c *ChannelGraph) UpdateEdgePolicy(edge *ChannelEdgePolicy) error {
 	c.cacheMu.Lock()
 	defer c.cacheMu.Unlock()
 
+	var isUpdate1 bool
 	err := c.db.Update(func(tx *bbolt.Tx) error {
-		return updateEdgePolicy(tx, edge)
+		var err error
+		isUpdate1, err = updateEdgePolicy(tx, edge)
+		return err
 	})
 	if err != nil {
 		return err
 	}
 
-	c.rejectCache.remove(edge.ChannelID)
-	c.chanCache.remove(edge.ChannelID)
+	if entry, ok := c.rejectCache.get(edge.ChannelID); ok {
```

Comment:
Better to just remove IMO, we have no guarantees that the updated policy doesn't change the value of the `rejectFlags`.

Reply:
the reject flags are only changed wrt to channel existence/zombie pruning. updating the edge policy shouldn't modify `rejectFlags`